kore/src/Kore/Step/Simplification/Condition.hs

```diff
@@ -131,3 +188,166 @@ simplifyPredicate sideCondition predicate = do
             [ "Expecting a \\top term, but found:"
             , unparse conditional
             ]
+
+simplifyConjunctions
+    :: InternalVariable variable
+    => Conditional variable any
+    -> Changed (Conditional variable any)
```

Comment:
Why does this operate on a `Conditional` when it ignores `term` and `substitution`?

Reply:
In `simplify` we operate on `Conditional`s, and I thought it would be more succinct and easier to read this way. Should I just inline the `Lens.traverseOf ...`? Or is the name `simplifyConjunctions` confusing, because it will only simplify the conjunctions from the `predicate`?